File: Backend/src/main/java/com/example/Java_Backend/entities/CartItem.java
package com.example.Java_Backend.entities;

import jakarta.persistence.*;
import lombok.Data;
import java.util.List;

@Data
public class CartItem {
    private Integer Id;

    private Integer CartId;

    private Integer ProductId;

    private Integer Quantity;

    private String FlavorNotes;

    private Integer GrindingOptionId;

    private String Weight;

    private Product Product;
}

File: KHE_AuthService/KHE_AuthService/Entities/CartItem.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace KHE_AuthService.Entities
{
    public class CartItem
    {
        public int Id { get; set; }

        public int CartId { get; set; }

        public int ProductId { get; set; }

        public int Quantity { get; set; }

        public int? GrindingOptionId { get; set; }

       
        public string? FlavorNotes { get; set; }

       
        public virtual Cart Cart { get; set; } = null!;

        public virtual GrindingOption? GrindingOption { get; set; }

       
        public virtual Product Product { get; set; } = null!;
    }
}
